src/server/rdb_save.cc

```diff
@@ -436,7 +436,7 @@ error_code RdbSerializer::SaveHSetObject(const PrimeValue& pv) {
     CHECK_EQ(kEncodingListPack, pv.Encoding());
 
     uint8_t* lp = (uint8_t*)pv.RObjPtr();
-    RETURN_ON_ERR(SaveListPackAsZiplist(lp));
+    RETURN_ON_ERR(SaveListPack(lp));
```

Comment:
We save an allocation here because `SaveListPackAsZipList` needs to create a new zip list and copy over the elements. Redis does not do this and so do we :)

Reply:
Redis changed his major version from 6 to 7. We do not do that now